stdlib/public/core/StringCharacterView.swift

```diff
@@ -439,7 +447,8 @@ extension String.CharacterView {
     let unicodeScalarRange =
       bounds.lowerBound._base..<bounds.upperBound._base
     return String.CharacterView(
-      String(_core).unicodeScalars[unicodeScalarRange]._core)
+      UnicodeScalarView(_core, coreOffset: _coreOffset)[unicodeScalarRange]._core,
+      coreOffset: bounds.lowerBound._base._position)
```

Comment:
Updated in the latest commit—I'd forgotten `CharacterView` already has a `unicodeScalars` property, which simplifies this a lot. Look okay?